aws/templates/resource/resource_s3.ftl

```diff
@@ -88,12 +88,20 @@
     ]
 [/#function]
 
-[#function getS3WebsiteConfiguration index error ]
-    [#return 
+[#function getS3WebsiteConfiguration index error redirectTo="" redirectProtocol="http"]
+    [#return
         {
             "IndexDocument" : index
         } +
-        attributeIfContent("ErrorDocument", error)
+        attributeIfContent("ErrorDocument", error) +
+        attributeIfContent(
+            "RedirectAllRequestsTo",
+            redirectTo,
+            {
+              "HostName" : redirectTo,
+              "Protocol" : redirectProtocol
```

Comment:
Might be worth making Protocol an optional configuration i.e. if_content add the protocol.

The default config is to redirect on the protocol that the request came in on which would be what we want for domain to new domain redirects sine you could either come via HTTP or HTTPS

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-s3-websiteconfiguration-redirectallrequeststo.html

Its worded kind of weird though..